lib/modules/notifications.bash

```diff
@@ -0,0 +1,251 @@
+                operation_header="🚀 *ArgoCD Deployment Success*"
+                status="Deployment successful"
+                from_label="Previous"
+                to_label="Current"
+                footer="Deployment completed successfully and application is healthy."
+                ;;
+            "deployment_failed_auto")
+                operation_header="❌ *ArgoCD Deployment Failed*"
+                status="Deployment failed - Auto rollback in progress"
+                from_label="Current"
+                to_label="Target"
+                footer="Automatic rollback initiated..."
+                ;;
+            "deployment_failed_manual")
+                operation_header="❌ *ArgoCD Deployment Failed*"
+                status="Deployment failed - Manual decision required"
+                from_label="Current"
+                to_label="Target"
+                footer="Manual rollback decision required on pipeline."
+                ;;
+            "rollback_success_auto")
+                operation_header="🔄 *ArgoCD Rollback Success*"
+                status="Auto rollback successful"
+                from_label="From"
+                to_label="To"
+                footer=""
+                ;;
+            "rollback_success_manual")
+                operation_header="🔄 *ArgoCD Rollback Success*"
+                status="Manual rollback successful"
+                from_label="From"
+                to_label="To"
+                footer=""
+                ;;
+            "rollback_failed_auto")
+                operation_header="❌ *ArgoCD Rollback Failed*"
+                status="Auto rollback failed"
+                from_label="From"
+                to_label="Target"
+                footer="Manual investigation required. Check logs for details."
+                ;;
+            "rollback_failed_manual")
+                operation_header="❌ *ArgoCD Rollback Failed*"
+                status="Manual rollback failed"
+                from_label="From"
+                to_label="Target"
+                footer="Manual investigation required. Check logs for details."
+                ;;
+        esac
+        
+        notification_message=$(create_notification_template "$app_name" "$status" "$from_label" "$from_revision" "$to_label" "$to_revision" "$footer" "$operation_header")
+        
+        # Escape the message for YAML (replace newlines with \n) - Cross-platform compatible
+        local escaped_message
+        escaped_message=$(printf '%s' "$notification_message" | awk '{printf "%s\\n", $0}' | sed 's/\\n$//')
+        
+        # Inject notification step using Buildkite's native Slack integration
+        local notification_pipeline
+        notification_pipeline=$(cat <<-EOF
+steps:
+  - label: ":slack: ArgoCD Plugin Notification"
+    command: "echo 'Sending notification to Slack...'"
+    notify:
+      - slack:
+          channels:
+            - "$slack_channel"
+          message: "$escaped_message"
```

Comment:
With Clusters and Queues being a thing, uploading Steps with/without `agents` attributes may not work as there might not be "default" queue to fall back on that could run this Step.